controllers/vrg_volrep.go

```diff
@@ -1622,30 +1672,31 @@ func (v *VRGInstance) fetchAndRestorePV(result *ctrl.Result) error {
 			continue
 		}
 
-		var pvList []corev1.PersistentVolume
+		var pvCount, pvcCount int
 
-		pvList, err = downloadPVs(objectStore, v.s3KeyPrefix())
+		// Restore all PVs found in the s3 store. If any failure, the next profile will be retried
+		pvCount, err = v.restorePVsFromObjectStore(objectStore, s3ProfileName)
 		if err != nil {
-			v.log.Error(err, fmt.Sprintf("error fetching PV cluster data from S3 profile %s", s3ProfileName))
-
 			continue
 		}
 
-		v.log.Info("Found PVs in s3 store", "count", len(pvList), "profile", s3ProfileName)
+		// Attempt to restore all PVCs from this profile. If a PVC is missing from this s3 stores or fails
+		// to restore, there will be a warning, but not a failure (no retry). In such cases, the PVC may be
+		// created when the application is created and it will bind to the PV correctly if the PVC name
+		// matches the PV.Spec.ClaimRef. However, the downside of tolerating failure is if an operator like
+		// CrunchyDB is responsible for creating and managing the lifecycle of their own PVCs, a newly created
+		// PVC may cause a new PV to be created.
+		// Ignoring PVC restore errors helps with the upgrade from ODF-4.12.x to 4.13
+		pvcCount, err = v.restorePVCsFromObjectStore(objectStore, s3ProfileName)
```

Comment:
(future) Restoring a PVC before its PV may cause a new volume to be provisioned, we need to backup PVCs and PVs as a single s3 PUT and/or ensure that a PVC restore has a matching PV that was restored, and error if not.